research/spoofax-org-export.tex

```diff
@@ -0,0 +1,234 @@
+one asks "What is the syntax of my language?". From such a description
+in a meta-language, the tokenizer and parser can be derived, without
+the designer of the language ever having to care about its
+implementation.
+
+This section goes over the parts that make up the specification of a
+language, and for each part the relevant part of Spoofax is
+given\footnote{This section follows the structure of the
+language specification portion of the compiler construction course at
+the TU Delft. The slides can be found here:
+\url{http://tudelft-in4303.github.io/lectures/specification/}.}. The parts
+of a language specification are:
+\begin{enumerate}
+\item \hyperref[sec-syntax-def]{Syntax Definition}: Specifying the syntax of a language.
+\item \hyperref[sec-static-analysis]{Static Semantics}: Describing the static analysis part of a
+language: type checking, name binding and variable scoping.
+\item \hyperref[sec-term-rewrite]{Term Rewriting and Program Transformation}: Rewriting ASTs to new
+ASTs, for example to declare desugaring rules.
+\item \hyperref[sec-dynamic-semantics]{Dynamic Semantics}: Defining what the language does upon execution.
+\end{enumerate}
+
+\subsection{Syntax Definition}
+\label{sec-syntax-def}
+The first part of the specification of a language is its syntax. The
+syntax of a language is often specified by means of a \emph{lexical
+grammar} and a \emph{context-free grammar}, as can be seen in the
+specification of, for example, Standard ML~\cite{Milner97}. The
+lexical grammar is most often defined using regular expression. It
+defines the individual words made up of characters, such as
+identifiers and numeric constants. The context-free grammar then
+defines syntactically valid sentences made up of words.
+
+\subsubsection{SDF3: syntax definition in Spoofax}
+\label{sec:orgheadline1}
+To specify a syntax definition declaratively in Spoofax, a DSL called
+\emph{SDF3}~\cite{Vollebregt12} is used, the third generation of the
+\emph{Syntax Definition Formalism} (SDF)~\cite{Heering89}. SDF3 uses
+context-free grammer productions for the specification of both the
+lexical syntax and the context-free syntax, a feature that was
+introduced in SDF2~\cite{Visser97}.
+
+The declarative nature of SDF3 allows for thinking in terms of the
+structure (the \emph{what}), instead of in terms of parser algorithms (the
+\emph{how}) as is the case with many current parsing
+algorithms~\cite{Kats10b}. The syntax definition can be used to
+make parsers which parse a textual representation to an AST, and
+pretty-printers for mapping ASTs back to text. However, due to its
+declarative nature, SDF3 is not limited to generating parsers and
+pretty printers: it can also be used for error recovery
+rules~\cite{deJonge12}, syntax highlighting rules and folding
+rules for editors.
+
+All of the other parts of Spoofax use the AST that is produced by a
+parser generated from a SDF3 specification.
```

Comment:
I guess it depends on how you read SDF; personally I would say "es dee ef", in which case it does start with a vowel.